DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/ManageCourseController.cs

```diff
@@ -303,27 +304,35 @@ private int GetIntegerFromStringOrConvertToZeroIfNull(string? input)
             EditCourseDetailsFormData formData
         )
         {
-            if (customisationName.Length > 250)
+            if (customisationName == string.Empty && courseService.DoesCourseNameExistAtCentre(
+                customisationId,
+                customisationName,
+                centreId,
+                formData.ApplicationId
+            ))
             {
                 ModelState.AddModelError(
-                    nameof(EditCourseDetailsViewModel.CustomisationNameSuffix),
+                    nameof(EditCourseDetailsViewModel.CustomisationName),
+                    "A course with this name already exists, add on to the course name to make it unique"

```

Comment:
I can't reply to the original comment any more for some reason, so for reference you said:
>The reason for this error message is because if the user deletes the customisation name and there is already an course with the same application name and no customisation name, it is then required to specify a customisation name. It looked a bit weird to me to just say "a course with this name already exists" when the box the error is highlighting has nothing in it. The course name it's talking about is the uneditable application name. Can you think of a better way to phrase it or do you think it wouldn't be confusing?

I personally don't think it would be confusing, to me an error message on an empty box saying "something with this name already exists" means something already has the name "" and so I need to think of a different one, but that might be me being a computer scientist where the empty string is just as valid as any other.
What I do know is that less is more when it comes to error messages. If it's more than a few words and if it takes any effort to understand then users are going to ignore it. I think the error message as-is breaks both these rules. 
How about something like "A course with no add on already exists"? I think that's much more clear than saying "a course with this name already exists" on an empty input. I think ideally we'd get some user testing on this (or at least someone at HEE's opinion) either way.

Reply:
That makes a lot of sense. I like "A course with no add on already exists", I've changed it to that :)